src/plugins/data_source/server/client/configure_client.ts

```diff
@@ -50,13 +50,29 @@ export const getDataSource = async (
 
 export const getCredential = async (
   dataSource: SavedObject<DataSourceAttributes>,
-  cryptographyClient: CryptographyClient
+  cryptography: CryptographyServiceSetup
 ): Promise<UsernamePasswordTypedContent> => {
+  const { endpoint } = dataSource.attributes!;
+
   const { username, password } = dataSource.attributes.auth.credentials!;
-  const decodedPassword = await cryptographyClient.decodeAndDecrypt(password);
+
+  const { decryptedText, encryptionContext } = await cryptography
+    .decodeAndDecrypt(password)
+    .catch(() => {
+      throw new Error(
+        'Encrypted "auth.credentials.password" contaminated. Please delete and create another data source.'
+      );
+    });
```

Comment:
The overwrite was done on purpose because the original error when decryption fails is not distinguishable from other encryption issues. But I agree with you because improving the SDK's error message is the right thing to do.